block-producer/src/block.rs

```diff
@@ -19,7 +19,7 @@ use crate::store::BlockInputsError;
 pub struct Block {
     pub header: BlockHeader,
     pub updated_accounts: Vec<UpdatedAccount>,
-    pub created_notes: Vec<(usize, usize, NoteEnvelope)>,
+    pub created_notes: Vec<(usize, usize, NoteCreated)>,
```

Comment:
If we put `NoteCreated` here, we don't need to provide additionally `batch_index` and `note_index` here.